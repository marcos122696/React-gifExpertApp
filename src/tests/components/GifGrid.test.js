import React from 'react'
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componenete <GifGrid />', () => {
    
    const category = 'Los Angeles';
    // const wrapper = shallow(<GifGrid category={category}/>);

    test('Debe de mostrar <GifGrid /> correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        const h3 = wrapper.find('h3');
        expect(wrapper).toMatchSnapshot();
        expect( h3.text() ).toBe( `Resultados de: ${ category }` );
    });

    test('Debe de mostrar items cuando se cargan imagenes', () => { 

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/gif.jpg',
            title:'titulo gif',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>); 
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
}); 
