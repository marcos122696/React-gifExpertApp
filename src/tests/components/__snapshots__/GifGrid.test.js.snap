// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componenete <GifGrid /> Debe de mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="text-result animate__animated animate__lightSpeedInLeft"
  >
    Resultados de: 
    Los Angeles
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
  <p
    className="text-no-result animate__animated animate__headShake"
  >
    No hay resultados
  </p>
</Fragment>
`;
